### PR TITLE
Issue #3498: ImportCheckControl refactored

### PR DESCRIPTION
**Pull request by kazachka**
Issue #3498 

inPkg renamed to packageName because package is keyword.

### CODE REVIEW

**Comment by jochenvdv** [comment]
PkgControl was renamed to ImportControl, so `currentPkgControl` should probably be `currentImportControl`.

**Comment by romani** [comment]
I agree with @jochenvdv.
Please do proposed renaming.

**Comment by romani** [comment]
Refactoring of code should result in no changes in  UT. Please explain reason of changes.

**Comment by kazachka** [comment]
I replaced package name AST node initialization to getPackageText. After this replacing visitToken hasn't AST that represents name of package, so I call log with PACKAGE_DEF AST node and update test to check warning on 1st column.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java [hunk 1]
```diff
@@ -237,7 +236,7 @@ public void testUrlBlank() throws Exception {
     public void testUrlNull() throws Exception {
         final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
         checkConfig.addAttribute("url", null);
-        final String[] expected = {"1:47: " + getCheckMessage(MSG_MISSING_FILE)};
+        final String[] expected = {"1:1: " + getCheckMessage(MSG_MISSING_FILE)};
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java]: I am ok with this update. Column value does not matter in this case.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlRegExpTest.java [hunk 2]
```diff
@@ -54,33 +54,33 @@ public void testLocateFinest() {
     @Test
     public void testCheckAccess() {
         assertEquals(AccessResult.DISALLOWED, icCommon.checkAccess(
-                "org.springframework.something",
-                "com.kazgroup.courtlink.common"));
+                "com.kazgroup.courtlink.common",
+                "org.springframework.something"));
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlRegExpTest.java]: ok, this update is due to change of arguments in method.
```
-    public AccessResult checkAccess(final String forImport, final String inPkg) {
+    public AccessResult checkAccess(final String inPkg, final String forImport) {
```

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java [hunk 3]
```diff
@@ -147,6 +136,34 @@ else if (currentLeaf != null) {
...
 
     /**
+     * Returns package text.
+     * @param ast PACKAGE_DEF ast node
+     * @return String that represents full package name
+     */
+    public String getPackageText(DetailAST ast) {
+        final DetailAST nameAST = ast.getLastChild().getPreviousSibling();
+        return FullIdent.createFullIdent(nameAST).getText();
+    }
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java]: please make it as static.
FYI: sad, it should be caught automatically - sevntu-checkstyle/sevntu.checkstyle#494

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java [hunk 4]
```diff
@@ -147,6 +136,34 @@ else if (currentLeaf != null) {
...
+
+    /**
+     * Returns import text.
+     * @param ast ast node that represents import
+     * @return String that represents importing class
+     */
+    private String getImportText(DetailAST ast) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java]: please make it static.

Comment (kazachka) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java]: Updated

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java [hunk 5]
```diff
@@ -147,6 +136,34 @@ else if (currentLeaf != null) {
     }
 
     /**
+     * Returns package text.
+     * @param ast PACKAGE_DEF ast node
+     * @return String that represents full package name
+     */
+    public static String getPackageText(DetailAST ast) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java]: Why `public`? I prefer to be very strict with modifiers.
Unless there is some reason, which I don't see, this should be `private`.

Comment (kazachka) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java]: Updated

**Comment by romani** [comment]
my points are addressed. Ok to merge.
@rnveach , your point is also fixed, please review one more time and merge PR.

**Comment by rnveach** [comment]
Merged.

**Pull request merged**